src/WordPress/Blueprints/Model/DataClass/UnzipStep.php

```diff
@@ -2,60 +2,64 @@
 
 namespace WordPress\Blueprints\Model\DataClass;
 
-class UnzipStep implements StepDefinitionInterface
-{
+class UnzipStep implements StepDefinitionInterface {
+
 	public const DISCRIMINATOR = 'unzip';
 
 	/** @var Progress */
-	public $progress;
+	public Progress $progress;
 
 	/** @var bool */
-	public $continueOnError;
+	public bool $continue_on_error;
```

Comment:
This is a WordPress coding standards-motivated change. But model data classes are derived from the scheme, so...   @adamziel does this mean we either change the coding standard or the scheme? As I understand, v2 is still in development...

Reply:
Analyzing the CompiledStep class:

```
class CompiledStep {
  public function __construct (
    public StepDefinitionInterface $step,
    public StepRunnerInterface $runner,
    public Tracker $progressTracker,
  ) { }
}
```

What is the reasoning behind `$progress` and `$continue_on_error` being declared at individual step level? 

```
/** @var Progress */
public $progress;

/** @var bool */
public $continue_on_error;
```

For `$progress` I wonder if this is something that should be refactored as the `Tracker` object is now external to the `StepDefinition`? It is passed directly to the runner. Although both come from the `CompiledStep`. Why won't the run method accept a single argument implementing the `CompiledStep`? It this due to hermetization?

For `$continue_on_error`, I would imagine it could be a field declared at an abstract level. Although, complicating the inheritance structure for a single field does not seem wise. So maybe only a getter should be declared at interface level?

Reply:
I would exclude the autogenerated code from PHPCS standards checks, only keep the PHP 7.0 compat checks around. Otherwise we'll keep running into annoying warnings that will require iterating on the code generator – there isn't much value in that.

Reply:
> What is the reasoning behind $progress and $continue_on_error being declared at individual step level?

`$progress` doesn't represent progress tracking but the progress part of the step declaration:

```js
{
	steps: [
		{
			step: 'defineWpConfigConsts',
			progress: {
				weight: 5,
				caption: 'Defining WordPress constants',
			}
		}
	]
}
```

About progress tracking:

BlueprintCompiler is what calculates the progress weight of each step and resource. If total progress is 100%, some steps may contribute 15% while others may contribute 27%. Each step reports its own progress, too. So from the perspective of a specific step, its own progress goes between 0% and 100%. It might report 10%, then 72%, then 100%. The top-level tracker reconciles that with that step's weight to calculate the total progress. Trackers are also hierarchical, so a top-level tracker may split into two sub-trackers, each representing 50% of the total progress, and each sub-tracker may split again. See the original TypeScript implementation for more context:

https://github.com/WordPress/wordpress-playground/blob/f9379b3fb26383a9ff874764ea1d27c95e994a56/packages/php-wasm/progress/src/lib/progress-tracker.ts#L72

---

About `$continue_on_error`, here's more context about its role:

* https://github.com/WordPress/wordpress-playground/issues/600
* https://github.com/WordPress/wordpress-playground/issues/604
* https://github.com/WordPress/wordpress-playground/pull/605

>  I would imagine it could be a field declared at an abstract level

Perhaps, I'm not that concerned about implementation details at this point. Let's leave it as it is for now and see what kind of abstractions emerge later on.